subsys/net/ip/dhcpv4.c

```diff
@@ -1166,15 +1167,17 @@ void net_dhcpv4_start(struct net_if *iface)
 		 */
 		iface->config.dhcpv4.xid = entropy;
 
-
-		/* RFC2131 4.1.1 requires we wait a random period
-		 * between 1 and 10 seconds before sending the initial
-		 * discover.
-		 */
-		timeout = entropy %
-				(CONFIG_NET_DHCPV4_INITIAL_DELAY_MAX -
-				  DHCPV4_INITIAL_DELAY_MIN) +
-				DHCPV4_INITIAL_DELAY_MIN;
+		/* Use default */
+		if (delay < 0) {
```

Comment:
I was thinking more of a providing an exact delay value in seconds in this parameter, but this apporach should also work. I would however convert it to `bool` type, IMO there's no need to use ambiguous integer values here.

Reply:
Yeah, but as entropy is needed for both delay and XID, this seemed easier.